modules/parser/src/main/scala/io/chrisdavenport/cormorant/parser/package.scala

```diff
@@ -21,8 +21,35 @@ package object parser {
     CSVParser.header.parseOnly(text).either.leftMap(ParseFailure.apply)
 
   def parseRows(text: String): Either[ParseFailure, CSV.Rows] =
-    CSVParser.fileBody.parseOnly(text).either.leftMap(ParseFailure.apply)
+    CSVParser.fileBody
+    .parseOnly(text)
+    .either
+    .leftMap(ParseFailure.apply)
+    .map { 
+      case rows@CSV.Rows(CSV.Row(x) :: _) => 
+        if (x.size > 1) filterLastRowIfEmpty(rows)
+        else rows
+      case otherwise => otherwise
+    }
 
   def parseComplete(text: String): Either[ParseFailure, CSV.Complete] =
-    CSVParser.`complete-file`.parseOnly(text).either.leftMap(ParseFailure.apply)
+    CSVParser.`complete-file`
+    .parseOnly(text)
+    .either
+    .leftMap(ParseFailure.apply)
+    .map{ case c@CSV.Complete(h@CSV.Headers(headers), rows) => 
+      // Due to The Grammar Being Unclear CRLF can and will be parsed as 
+      // a field. However the specification states that each
+      if (headers.size > 1) {
+        CSV.Complete(h, filterLastRowIfEmpty(rows))
+      } else {
+        c
+      }
+    }
+  private def filterLastRowIfEmpty(rows: CSV.Rows): CSV.Rows = {
+    rows.rows.reverse match {
```

Comment:
Is the model a `List` ? How often would this function get called? I'm worried about this doing more work than needed.